tools/module_get_cubed_sphere_inc.F90

```diff
@@ -0,0 +1,551 @@
+#define NC_ERR_STOP(status) \
```

Comment:
I believe most if not all of the I/O code in this file can be replaced with standard FMS calls. This could be done similar to that in `get_nggps_ic` which also reads in native-grid data.

Reply:
@lharris4 If I recall correctly, FMS calls can only be used to read FMS-formatted restart files. Our aim with JEDI in the GDASApp and Global Workflow is to read cubed-sphere history files.

Reply:
Lucas is correct that FMS can read FMS and non-FMS created NetCDF files on the native grid or for any grid.  Otherwise, our data_override capability to read in standardized forcing field datasets would not work.

For non-native grid NetCDF files and to keep each MPI-rank from reading in the full dataset, you can use:
- use fms2_io::read_data to ingest the lat-lon data describing the fields in the file
- determine indices that define a bounding box for the MPI-rank (with margins for interpolation)
- use fms2_io::read_data to ingest only the portion of the field(s) within the bounding box indices
- interpolate data to the native grid using a method defined in horiz_interp

For a native grid NetCDF defined on the cubed-sphere, one can use the fms2_io::read_data to read in the full field or if you know your particular indices you can read in a subset of the data.  I believe you can easily use the restart functionality as well without any issues.

Reply:
@bensonr Thanks for the info, I will look into changes using this functionality